format/midi/midi.go

```diff
@@ -0,0 +1,222 @@
+package midi
+
+// https://www.midi.org/specifications/item/the-midi-1-0-specification
+
+import (
+	"bytes"
+	"embed"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/interp"
+)
+
+type context struct {
+	tick    uint64
+	running uint8
+	casio   bool
+}
+
+//go:embed midi.md
+var midiFS embed.FS
+
+func init() {
+	interp.RegisterFormat(
+		format.MIDI,
+		&decode.Format{
+			Description: "Standard MIDI file",
+			Groups:      []*decode.Group{format.Probe},
+			DecodeFn:    decodeMIDI,
+		})
+
+	interp.RegisterFS(midiFS)
+}
+
+func decodeMIDI(d *decode.D) any {
+	d.Endian = decode.BigEndian
+
+	// ... decode header
+	println(">> 1")
+	if err := skipTo(d, "MThd"); err != nil {
```

Comment:
Midi files can have some unknown header that should be skipped?

If it's possible to accurately detect it's midi the format could be in the probe group

Reply:
The MIDI spec technically allows a MIDI file to include _other_ chunks but TBH I've never seen one that didn't start with the header. 

I've reworked the decoder to expect an _MThd_ (header) chunk as the first chunk so that it will fail fast when probing. It will still skip _unknown_ chunks after the header though.

Reply:
Aha ok, i think the CI error in thmb.mp4 was probably because midi is in the probe (i didn't notice it was) and somehow that mp4 box content got successfully decoded as midi, can happen if the decode function is a bit too liberal and allows empty or no "entries" to be decoded.

Reply:
Fixed in [ea3e0898bf88241fc3a6ef936b4e72bc89018ed7](https://github.com/wader/fq/pull/1004/commits/ea3e0898bf88241fc3a6ef936b4e72bc89018ed7)